Sources/TuistGraph/Models/TargetDependency.swift

```diff
@@ -17,3 +17,103 @@ public enum TargetDependency: Equatable, Hashable {
     case cocoapods(path: AbsolutePath)
     case xctest
 }
+
+// MARK: - Codable
+
+extension TargetDependency {
+    private enum Kind: String, Codable {
+        case target
+        case project
+        case framework
+        case xcFramework
+        case library
+        case package
+        case sdk
+        case cocoapods
+        case xctest
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case kind
+        case name
+        case target
+        case path
+        case publicHeaders
+        case swiftModuleMap
+        case product
+        case status
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let kind = try container.decode(Kind.self, forKey: .kind)
+        switch kind {
+        case .target:
+            let name = try container.decode(String.self, forKey: .name)
+            self = .target(name: name)
+        case .project:
+            let target = try container.decode(String.self, forKey: .target)
+            let path = try container.decode(AbsolutePath.self, forKey: .path)
+            self = .project(target: target, path: path)
+        case .framework:
+            let path = try container.decode(AbsolutePath.self, forKey: .path)
+            self = .framework(path: path)
+        case .xcFramework:
+            let path = try container.decode(AbsolutePath.self, forKey: .path)
+            self = .xcFramework(path: path)
+        case .library:
+            let path = try container.decode(AbsolutePath.self, forKey: .path)
+            let publicHeaders = try container.decode(AbsolutePath.self, forKey: .publicHeaders)
+            let swiftModuleMap = try container.decode(AbsolutePath?.self, forKey: .swiftModuleMap)
```

Comment:
I think for optionals it's more clear to use `encodeIfPresent` and `decodeIfPresent` (this same comment applies for multiple files in this PR)